diff --git a/deploy/Dockerfile b/deploy/Dockerfile
index b631963a..06fbe31d 100644
--- a/deploy/Dockerfile
+++ b/deploy/Dockerfile
@@ -49,6 +49,8 @@ RUN go install github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/doc
 # Get ACR docker env credential helper
 RUN go install github.com/chrismellard/docker-credential-acr-env
 
+RUN mkdir -p /kaniko && chmod 777 /kaniko
+
 RUN \
   --mount=type=cache,target=/root/.cache/go-build \
   --mount=type=cache,target=/go/pkg \
@@ -64,7 +66,7 @@ FROM busybox:musl AS busybox
 FROM scratch AS kaniko-base-slim
 
 # Create kaniko directory with world write permission to allow non root run
-RUN --mount=from=busybox,dst=/usr/ ["busybox", "sh", "-c", "mkdir -p /kaniko && chmod 777 /kaniko"]
+COPY --from=builder /kaniko /kaniko
 
 COPY --from=certs /etc/ssl/certs/ca-certificates.crt /kaniko/ssl/certs/
 COPY files/nsswitch.conf /etc/nsswitch.conf
